### DISCUSSION TITLE
control easyeffects from command line

### DISCUSSION THREAD

**Discussion by jtharkey** [wwmm/easyeffects#1416] [Q&A]
is it possible to have easyeffects (actually, i'm still on pulseeffects) be controlled live, in real-time, from the command line? i know i can change settings with gsettings, but that doesn't seem to update in real-time. i also know i can load presets but that doesn't seem to do it either. i'm stumpted.

i have a graphical EQ UI that isn't the easyeffects UI. i can't use that in this application. i want to "marry" that to easyeffects so that the eq changes as the user interacts with my UI. am i just using it wrong? i do run it with --qapplication-service

**Comment by wwmm** [comment]
>i know i can change settings with gsettings, but that doesn't seem to update in real-time. i also know i can load presets but that doesn't seem to do it either. i'm stumpted.

All our settings are handled through GSettings. When you set a parameter in our window what the widget is doing is changing a gsettings key. And that is forward by gsettings to the code that is actually applying effects. Same thing when loading a preset. What the related classes do is updating the GSettings database. I do not notice a delay in this task.

>i have a graphical EQ UI that isn't the easyeffects UI. i can't use that in this application. i want to "marry" that to easyeffects so that the eq changes as the user interacts with my UI

All you have to do is using GSettings binding for QT and binding your widgets to them.

**Comment by jtharkey** [comment]
Thank you for the reply. I'll try using the QT bindings. But you're saying that using "gsettings set ..." alone should do it? I wonder what I'm doing wrong.

-------- Original Message --------

**Reply by wwmm** [reply]
>But you're saying that using "gsettings set ..." alone should do it?

Yes. Assuming that the path to the key is correct the idea is the same as the one I explain at https://github.com/wwmm/easyeffects/discussions/916.

**Comment by jtharkey** [comment]
Understood. Thank you.

-------- Original Message --------
